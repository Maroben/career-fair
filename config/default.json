{
    "jwtPrivateKey": "",
    "db": "",
    "port": 3000,
    "endpoint": "http://localhost:3000/api"
}
