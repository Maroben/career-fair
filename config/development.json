{
    "db": "mongodb://localhost:27017/career-dev",
    "port": 3800,
    "endpoint": "http://localhost:3800/api"
}
