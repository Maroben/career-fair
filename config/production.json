{
    "db": "mongodb://localhost:27017/career-prod",
    "port": 8080,
    "endpoint": "http://hsrstellenboerse.ch/api"
}
