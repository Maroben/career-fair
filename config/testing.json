{
    "db": "mongodb://localhost:27017/career-test",
    "port": 3801
}
