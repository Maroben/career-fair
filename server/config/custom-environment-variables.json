{
	"jwtPrivateKey": "hsrcareer_jwtPrivateKey",
	"mongoDb": "hsrcareer_db",
	"mongoPort": "hsrcareer_db_port",
	"mongoUser": "hsrcareer_db_user",
	"mongoPw": "hsrcareer_db_pw"
}
