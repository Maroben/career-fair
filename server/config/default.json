{
	"jwtPrivateKey": "",
	"adminPw": "",
	"marobenPw": "",
	"userPw": "",
	"db": "mongodb://localhost:27017/career"
}
