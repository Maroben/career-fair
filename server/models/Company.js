const mongoose = require("mongoose")
const Joi = require("@hapi/joi")

const companySchema = new mongoose.Schema({
	name: {
		type: String,
		required: true,
		minlength: 2,
		maxlength: 32
	},
	location: {
		type: String,
		maxlength: 16
	},
	info: {
		type: String,
		maxlength: 512
	},
	description: {
		type: String,
		maxlength: 2056
	},
	subjects: {
		type: Array
	},
	employments: {
		type: Array
	},
	tags: {
		type: Array
	},
	links: {
		type: Object
	}
})

const companyJoi = {
	name: Joi.string()
		.min(2)
		.max(32)
		.required(),
	info: Joi.string().max(512),
	description: Joi.string().max(2056),
	subjects: Joi.array(),
	employments: Joi.array(),
	tags: Joi.array(),
	links: Joi.object()
}

const locationJoi = {
	location: Joi.string().max(16)
}

module.exports = {
	Model: mongoose.model("companies", companySchema),
	attr: [
		["name", "location", "info", "description", "subjects", "employments", "links", "tags"],
		["_id", "name", "location", "info", "subjects", "tags"]
	],
	validateCompany: (company) => {
		return Joi.validate(company, companyJoi)
	},
	validateLocation: (location) => {
		return Joi.validate(location, locationJoi)
	}
}
