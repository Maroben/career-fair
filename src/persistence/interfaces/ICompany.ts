import { Document } from "mongoose"

export default interface ICompany extends Document {
    name: string
    info: string
    description: string
    location: string
}
