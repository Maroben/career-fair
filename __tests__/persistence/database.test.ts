import mongoose from "mongoose"
import config from "config"
import _ from "lodash"
import Database from "../../src/persistence/Database"
import Company, { properties } from "../../src/persistence/models/CompanyModel"
import ICompany from "../../src/persistence/interfaces/ICompany"

describe("Database with Company", () => {
    beforeAll(async () => {
        await mongoose.connect(config.get("db"), {
            useNewUrlParser: true,
            useUnifiedTopology: true,
            useCreateIndex: true,
            useFindAndModify: false
        })
    })

    afterAll(async () => {
        await Company.deleteMany({})
        await mongoose.connection.close()
    })

    it("CRUD Company", async () => {
        const db = new Database<ICompany>(Company)

        expect(async () => await db.get({ _id: "fakeid" })).rejects

        const company1: ICompany = new Company({
            name: "Name1",
            info: "Info1",
            description: "Description1",
            location: "Location1"
        })
        const company2: ICompany = new Company({
            name: "Name2",
            info: "Info2",
            description: "Description2",
            location: "Location2"
        })

        const received1 = await db.post(company1)
        const received2 = await db.post(company2)
        expect(_.pick(received1, properties)).toMatchSnapshot()
        expect(_.pick(received2, properties)).toMatchSnapshot()

        const received3 = await db.getAll()
        expect(received3).toHaveLength(2)

        const received4 = await db.get(received2._id)

        expect(_.pick(received2, ["_id", ...properties])).toMatchObject(
            _.pick(received4, ["_id", ...properties])
        )

        const company5: ICompany = new Company({
            name: "Name5",
            info: "Info5",
            description: "Description5",
            location: "Location5"
        })

        const received5 = await db.put(company5, received1._id)
        expect(_.pick(received5, properties)).toMatchSnapshot()

        const received6 = await db.delete(received1._id)
        expect(received6).toMatchObject(received5)
        expect(async () => await db.get(received1._id)).rejects
    })
})
