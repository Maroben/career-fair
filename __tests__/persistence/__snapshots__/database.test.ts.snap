// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Database with Company CRUD Company 1`] = `
Object {
  "description": "Description1",
  "info": "Info1",
  "location": "Location1",
  "name": "Name1",
}
`;

exports[`Database with Company CRUD Company 2`] = `
Object {
  "description": "Description2",
  "info": "Info2",
  "location": "Location2",
  "name": "Name2",
}
`;

exports[`Database with Company CRUD Company 3`] = `
Object {
  "description": "Description5",
  "info": "Info5",
  "location": "Location5",
  "name": "Name5",
}
`;
